.github/workflows/ci.yaml

```diff
@@ -0,0 +1,97 @@
+# yamllint disable rule:line-length
+# yamllint disable rule:braces
+
+name: Continuous Integration
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 2.15
+      - 2.16
+      - 3.0
+
+jobs:
+  tests:
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      matrix:
+        operating-system:
+          - ubuntu-latest
+        php-version:
+          - 7.1
+          - 7.2
+          - 7.3
+          - 7.4
+
+        include:
+          - operating-system: 'ubuntu-latest'
+            php-version: 8.0
+            composer-flags: '--ignore-platform-req=php'
+            PHP_CS_FIXER_IGNORE_ENV: 1
+
+          - operating-system: 'ubuntu-latest'
+            php-version: 5.6
+            PHP_CS_FIXER_TEST_USE_LEGACY_TOKENIZER: 1
+            SYMFONY_DEPRECATIONS_HELPER: disabled
+
+          - operating-system: 'windows-latest'
+            php-version: 7.3
+            FAST_LINT_TEST_CASES: 1
+
+          - operating-system: 'windows-latest'
+            php-version: 5.6
+            SKIP_LINT_TEST_CASES: 1
+
+          - operating-system: 'macos-latest'
+            php-version: 7.4
+            PHP_CS_FIXER_FUTURE_MODE: 1
+
+    name: PHP ${{ matrix.php-version }} CI on ${{ matrix.operating-system }} ${{ matrix.composer-flags }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          tools: composer:v2
+
+      - name: Get Composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
```

Comment:
Let's replace `cache-files-dir` with `cache-dir`.

Reply:
Changed. 

Is it right that Composer can figure out it uses stale caches? I mean, if we'd cache everything, it might not ask Packagist if there's a new version. To be clear, I'm not so sure how this works.

Reply:
We always do `composer update`, so Composer will check for newer versions.